src/libstd/fs.rs

```diff
@@ -436,19 +436,19 @@ impl OpenOptions {
     /// Note that setting `.write(true).append(true)` has the same effect as
     /// setting only `.append(true)`.
     ///
-    /// For most filesystems the operating system guarantees all writes are
+    /// For most filesystems, the operating system guarantees all writes to be
     /// atomic: no writes get mangled because another process writes at the same
     /// time.
     ///
     /// One maybe obvious note when using append-mode: make sure that all data
-    /// that belongs together, is written the the file in one operation. This
+    /// that belongs together is written to the file in one operation. This
     /// can be done by concatenating strings before passing them to `write()`,
-    /// or using a buffered writer (with a more than adequately sized buffer)
+    /// or using a buffered writer (with a more than adequately-sized buffer),
```

Comment:
I don't believe a `-` is needed between an adverb and the adjective it modifies ("adequately-sized" doesn't look "correct" to me, in other words). However, this kind of "multi-word adjective" -- "more than adequately sized" -- always feels a little awkward and informal to me in any case. Perhaps "with a buffer of sufficient size" or "with a sufficiently large buffer". (Also, I have no idea what a "more than adequately sized buffer" even is supposed to mean -- surely "adequate" size would be enough, since that is sort of the definition of adequate?)